SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -280,6 +280,15 @@ - (void)main
 
 - (void)tryWithRemainingRetries:(NSUInteger)remainingRetries
 {
+    // If the message has been deleted, abort send.
+    if (![TSOutgoingMessage fetchObjectWithUniqueID:self.message.uniqueId]) {
```

Comment:
If we can detect that a user has deleted an outgoing message before the sender could send it, abort the send.

Reply:
This is a more desirable UX, I believe.